torch/csrc/jit/script/compiler.cpp

```diff
@@ -484,66 +484,73 @@ Value* tryMatchArgument(
   return value;
 }
 
+at::optional<size_t> findInputWithName(const std::string& name, at::ArrayRef<NamedValue> kwargs) {
+  for(size_t i = 0; i < kwargs.size(); ++i) {
+    if(kwargs[i].name() == name)
+      return i;
+  }
+  return at::nullopt;
+}
+
 at::optional<std::vector<Value*>> tryMatchSchema(
   const FunctionSchema& schema,
   const SourceRange& loc,
   Graph& graph,
-  at::ArrayRef<NamedValue> inputs,
-  at::ArrayRef<NamedValue> attributes,
+  at::ArrayRef<NamedValue> args,
+  at::ArrayRef<NamedValue> kwargs,
   std::ostream& failure_messages) {
     auto err = [&]() -> std::ostream& {
       failure_messages << "\nfor operator " << schema << ":\n";
       return failure_messages;
     };
 
-    std::vector<at::optional<NamedValue>> positional_inputs(schema.arguments.size(), at::nullopt);
+    std::vector<Value*> positional_inputs;
+    std::vector<bool> used_kwarg(kwargs.size(), false);
 
-    size_t total_inputs = attributes.size() + inputs.size();
-    if(total_inputs > schema.arguments.size()) {
-      err() << "expected at most " << schema.arguments.size() << " arguments "
-      << "but found " << total_inputs << "\n" << loc << "\n";
-      return at::nullopt;
-    }
-    // fill in positional arguments
-    for(size_t i = 0; i < inputs.size(); ++i) {
-      positional_inputs[i] = inputs[i];
-    }
-    // fill in named arguments
-    for(const NamedValue& nv : attributes) {
-      auto idx = schema.argumentIndexWithName(nv.name());
-      if(!idx) {
-        err() << "unknown keyword argument '" << nv.name() << "'\n" << nv.locOr(loc);
-        return at::nullopt;
-      }
-      if(positional_inputs[*idx]) {
-        err() << "argument " <<  nv.name() << " specified twice \n" << nv.locOr(loc);
-        return at::nullopt;
-      }
-      positional_inputs[*idx] = nv;
-    }
-    // fill in default values
-    for(size_t i = 0; i < positional_inputs.size(); ++i) {
-      if(positional_inputs[i])
-        continue;
-      auto default_value = schema.arguments[i].default_value;
-      if(!default_value) {
+    for(size_t i = 0; i < schema.arguments.size(); ++i) {
+      const auto& arg = schema.arguments[i];
+      at::optional<NamedValue> v;
+      if(i < args.size()) {
+        v = args[i];
+      } else if(auto idx = findInputWithName(arg.name, kwargs))  {
+        const NamedValue& nv = kwargs[*idx];
+        if(used_kwarg[*idx]) {
+          err() << "argument " << nv.name() << " specified twice \n" << nv.locOr(loc);
```

Comment:
Errr... isn't this an internal error in the schema? It means that the *schema* specified an argument twice?

Reply:
Oh yeah, I guess that it is.